### PR TITLE
CI Setup

### PR DESCRIPTION
**Pull request by rguo123**
Setup CI to run typescript check, eslint on changed files, and unit tests

----

> [!IMPORTANT]
> Sets up CI workflow to run ESLint, TypeScript checks, and unit tests on pull requests using GitHub Actions.
> 
>   - **CI Setup**:
>     - Adds `.github/workflows/ci.yml` to configure CI for pull requests.
>     - Runs on `ubuntu-latest`.
>   - **Linting and Testing**:
>     - Uses `actions/checkout@v4` for code checkout.
>     - Sets up Node.js 20 with `actions/setup-node@v4`.
>     - Caches `node_modules` using `actions/cache@v4`.
>     - Installs dependencies with `npm ci` if cache is not hit.
>     - Collects changed files with `git diff` and filters for `.js`, `.jsx`, `.ts`, `.tsx`.
>     - Runs `eslint` on changed files if any.
>     - Executes TypeScript checks with `npx tsc --emitDeclarationOnly`.
>     - Runs unit tests with `npm run test`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=nomic-ai%2Fdeepscatter&utm_source=github&utm_medium=referral)<sup> for 970b2ecb986d3ee160629922cc115ba7fe85970d. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by rguo123** [comment]
* **#171** <a href="https://app.graphite.dev/github/pr/nomic-ai/deepscatter/171?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rguo123 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: CI Testing
+on: pull_request
+permissions: write-all
```
[the remark this answers is not in the document]
Comment (rguo123) [reply .github/workflows/ci.yml:3]: fair enough

**Review by bmschmidt — APPROVED** [review]
TY!

**Comment by bmschmidt** [comment]
### Merge activity

* **Nov 5, 3:50 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/nomic-ai/deepscatter/171).

**Pull request merged**